### PR TITLE
Add citekey to searched fields

### LINKED ISSUE #46: Search by citekey

### PR DESCRIPTION
**Pull request by lockshaw**
Should resolve #46 as I understand it. Let me know if there's anything I missed/need to add!

### CODE REVIEW

**Review by alicangok — APPROVED** [review]

**Comment by alicangok** [comment]
(Preface: Possible alternative solution for the not-so-experts like myself)

Thank you. I'm not very familiar with the Obsidian plugin API, and it seems that in order to apply this change to my local obsidian client, I need to download this modified version to my MyVault/.obsidian/plugins folder, then compile using the NodeJS, right?

I figured (and please correct me if I'm wrong, or if this is not a healthy method), I could modify MyVault/.obsidian/plugins/obsidian-citation-plugin/main.js to achieve the functionality "add citekey to searched fields".

So I went ahead and changed line 14668 of main.js from
`        return item.title + " " + item.authorString + " " + item.year`
to 
`        return item.title + " " + item.authorString + " " + item.year + " ".id + item.id;`

**Comment by lockshaw** [comment]
> Thank you. I'm not very familiar with the Obsidian plugin API, and it seems that in order to apply this change to my local obsidian client, I need to download this modified version to my MyVault/.obsidian/plugins folder, then compile using the NodeJS, right?

For now, yes. Ideally if/once a maintainer approves and merges this an updated release can be made

**Comment by lockshaw** [comment]
Since there seems to be some general interest in this feature and I can't merge it here without a maintainer, I've setup a release with an improved version of this feature at https://github.com/lockshaw/obsidian-citation-plugin/releases/tag/0.4.5. @AdriaanRol @allcangok 

For now I'm going to close this as there doesn't appear to be a clear timeline for merging, but @hans if you eventually get time to take a look at it (no rush, totally get it :slightly_smiling_face:) let me know and I'd be happy to put a PR back together again.

**Pull request closed without merging**
